mlflow/utils/file_utils.py

```diff
@@ -689,18 +702,34 @@ def run_download(range_start, range_end):
         futures[thread_pool_executor.submit(run_download, range_start, range_end)] = i
 
     failed_downloads = {}
+    pbar = (
+        tqdm(
+            total=file_size,
+            unit="iB",
+            unit_scale=True,
+            unit_divisor=1024,
+            desc=f"Downloading file {download_path}",
+            miniters=1,
+        )
+        if _TQDM_AVAILABLE
+        else None
+    )
     for future in as_completed(futures):
         index = futures[future]
         try:
             result = future.result()
             if result is not None:
                 failed_downloads[index] = result
-
+            if pbar:
+                pbar.update(min(file_size - index * chunk_size, chunk_size))
+                pbar.refresh()
         except Exception as e:
             failed_downloads[index] = {
                 "error_status_code": 500,
                 "error_text": repr(e),
             }
+    if pbar:
+        pbar.close()
```

Comment:
Does tqdm support context manager?

Reply:
Yes it does, but we don't want to make tqdm as our dependency so I want to avoid coping the logic twice like:
```
if pbar: 
    with pbar: do_something
else:
    do_something
```